lib/logstash/pipeline.rb

```diff
@@ -252,6 +252,7 @@ def outputworker
   #
   # This method is intended to be called from another thread
   def shutdown
+    Thread.new { report_inflight_events(@input_to_filter, @filter_to_output, @outputs) }
```

Comment:
@jsvd +1 on being explicit. Maybe we compromise and name the reporter "SomethingSomethingTask" to indicate it's a separate task/thread? ShutdownReporterTask, or something?